uktrade/data-hub-frontend
# Feature/hook up dash chart to api

## Description of change

Hooks up the investment summary chart from the new dashboard page to the new API endpoint for investment summaries.

I have also refactored some of the chart component following the Single Responsibility Principal to make it easier to understand. 

This has meant splitting the chart into:

- A `BarChart` component that is purely responsible for rendering a chart
- A `DataTable` component that renders the data in a table
- A `DataSummary` which allows users to toggle between the `DataTable` and `BarChart` views for the data
- A `DataSummaryPicker` which allows the user to switch between data sets with the select drop down

![Data Summary Hierarchy](https://user-images.githubusercontent.com/1234577/109016070-b3080880-76ad-11eb-8315-475661d87398.png)

## Test instructions

With the layoutTesting flag set (or just temporarily edit 'src/apps/dashboard/views/dashboard.njk' so that it points to the new dashboad), visit the dashboard page and take a look at the chart. 

Choose between the two different years in the drop down. The chart should look the same as it did previously, but with data loaded from the API instead.

I have also made some amendments around storybook, so also check that the chart is showing correctly there.

There are now a storybooks for:

- `Chart/BarChart`
- `DataTable`
- `DataSummaryPicker` (this is taken from the former `Chart` stories)

## Screenshots

Pretty much the same as before, but with the data coming from the API:

![Screenshot from 2021-02-08 20-11-55](https://user-images.githubusercontent.com/1234577/107275620-ec077280-6a49-11eb-982d-b3ceee7043ed.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
